pkg/zonemgr/zone_file.go

```diff
@@ -2,180 +2,151 @@ package zonemgr
 }
 
-func (zoneFile *ZoneFile) setDefaultValues() {
-	zoneFile.domainName = fmt.Sprintf("%s.secondary", zoneFile.clusterName)
-
-	if zoneFile.userSubdomain != "" {
-		zoneFile.domainName = fmt.Sprintf("%s.%s", zoneFile.domainName, zoneFile.userSubdomain)
-	}
-
-	if zoneFile.userNameserverName == "" {
-		zoneFile.userNameserverName = fmt.Sprintf("ns1.%s", zoneFile.domainName)
-	}
-	/*
-		if zoneFile.userNameserverIP == "" {
-			//TODO default???
-		}
-	*/
-	if zoneFile.userAdminEmail == "" {
-		zoneFile.userAdminEmail = fmt.Sprintf("admin.%s", zoneFile.domainName)
+func (zoneFile *ZoneFile) initCustomFields() {
+	if zoneFile.domain == "" {
+		zoneFile.domain = domainDefault
+	} else {
+		zoneFile.domain = fmt.Sprintf("%s.%s", domainDefault, zoneFile.domain)
 	}
+	zoneFile.nameServerName = fmt.Sprintf("%s.%s", nameServerDefault, zoneFile.domain)
+	zoneFile.adminEmail = fmt.Sprintf("%s.%s", adminEmailDefault, zoneFile.domain)
 }
 
-/*
-$ORIGIN mycluster.secondary.mydomain.com
-$TTL 1750
-@       IN      SOA ns1.mycluster.secondary.mydomain.com admin@secdns.com (
-*/
 func (zoneFile *ZoneFile) generateHeaderPrefix() {
-	zoneFile.headerPref = fmt.Sprintf("\n$ORIGIN %s \n$TTL %s \n@ IN SOA %s %s (", zoneFile.domainName, ttl,
-		zoneFile.userNameserverName, zoneFile.userAdminEmail)
-
-	//fmt.Println("======= prefix:\n", zoneFile.headerPref)
-
+	zoneFile.headerPref = fmt.Sprintf("$ORIGIN %s. \n$TTL %s \n@ IN SOA %s. %s. (", zoneFile.domain, ttl,
+		zoneFile.nameServerName, zoneFile.adminEmail)
 }
 
-/*
-									7200       ; refresh (2 hours)
-	                                3600       ; retry (1 hour)
-	                                1209600    ; expire (2 weeks)
-	                                3600       ; minimum (1 hour)
-	                                )
-	        IN      NS  ns1.mycluster.secondary.mydomain.com
-	        IN      A   1.2.3.4
-*/
 func (zoneFile *ZoneFile) generateHeaderSuffix() {
 	zoneFile.headerSuf = fmt.Sprintf(" %s %s %s %s)\n", refresh, retry, expire, ttl)
 
-	if zoneFile.userNameserverName != "" {
-		zoneFile.headerSuf += fmt.Sprintf("IN NS %s\n", zoneFile.userNameserverName)
+	if zoneFile.nameServerName != "" {
+		zoneFile.headerSuf += fmt.Sprintf("IN NS %s.\n", zoneFile.nameServerName)
 	}
 
-	if zoneFile.userNameserverIP != "" {
-		zoneFile.headerSuf += fmt.Sprintf("IN A %s\n", zoneFile.userNameserverIP)
+	if zoneFile.nameServerIP != "" {
+		zoneFile.headerSuf += fmt.Sprintf("IN A %s\n", zoneFile.nameServerIP)
 	}
 }
 
 func (zoneFile *ZoneFile) generateHeader() (header string) {
 	return zoneFile.headerPref + strconv.Itoa(zoneFile.soaSerial) + zoneFile.headerSuf
 }
 
-//////////// maps impl
-
-func (zoneFile *ZoneFile) updateVMIRecords(vmiName string, vmiNamespace string, interfaces []v1.VirtualMachineInstanceNetworkInterface) (isUpdated bool, err error) {
+func (zoneFile *ZoneFile) updateVMIRecords(vmiName string, vmiNamespace string, interfaces []v1.VirtualMachineInstanceNetworkInterface) (isUpdated bool) {
```

Comment:
please dont use named return values, simply declare one and return it
(you already return it anyhow btw)

Reply:
done